src/Elasticsearch.Net/Diagnostics/Diagnostic.cs

```diff
@@ -45,8 +45,7 @@ internal set
 			}
 		}
 
-
 		//_source can be null if Default instance
-		public void Dispose() => _source?.StopActivity(SetEndTime(DateTime.UtcNow), EndState);
+		protected override void Dispose(bool disposing) => _source?.StopActivity(SetEndTime(DateTime.UtcNow), EndState);
```

Comment:
We did this here as well: https://github.com/elastic/elastic-transport-net/blob/main/src/Elastic.Transport/Diagnostics/Diagnostic.cs#L50

Wonder if we need to guard against `disposing` actually being `true`? 

I remember reading the base implementation and seeing this would only be called once, if this is true maybe we should add a comment about that here (and in `elastic-transport-net` for the next time we reread this line :D)

Reply:
Yes I copied this from transport! 😄 

I'll take a look into the base and see what needs to happen here. Should probably following the IDisposable pattern properly by checking the `bool`!

Reply:
I've pushed an update to properly implement the `IDisposable` pattern for this derived type.